### PR TITLE
[Objectfifo] Create a new pass to flatten vectorized ops

### PR DESCRIPTION
**Pull request by Abhishek-Varma**
-- This commit creates a new pass `iree-amdaie-flatten-vectorized-ops` which currently linearizes ONLY `arith.truncf` op.
-- This is the first commit that needs to go in to enable ["Matmul + truncf"](https://github.com/nod-ai/iree-amd-aie/pull/822).

Signed-off-by: Abhishek Varma <abhvarma@amd.com>

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFlattenVectorizedOps.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/IR/AMDAIEOps.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "mlir/Dialect/Vector/IR/VectorOps.h"
+#include "mlir/IR/IRMapping.h"
+#include "mlir/IR/Iterators.h"
```
Comment (jtuyls) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFlattenVectorizedOps.cpp:11]: Not used?
```suggestion
```

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFlattenVectorizedOps.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+class AMDAIEFlattenVectorizedOpsPass
+    : public impl::AMDAIEFlattenVectorizedOpsBase<
+          AMDAIEFlattenVectorizedOpsPass> {
+ public:
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<AMDAIEDialect, memref::MemRefDialect>();
+  }
+
+  AMDAIEFlattenVectorizedOpsPass() = default;
+  AMDAIEFlattenVectorizedOpsPass(const AMDAIEFlattenVectorizedOpsPass &pass){};
```
Comment (jtuyls) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFlattenVectorizedOps.cpp:28]: Can be removed.
```suggestion
```

**Review by jtuyls — APPROVED** [review]
LGTM.


> // TODO(avarma): Currently this is fixated on just `arith.truncf`. Follow-up
  //               on this later to generalize.

Ok for me for now, but yeah, we will need to discuss on how to generalize this well.

**Comment by yzhang93** [comment]
One minor title suggestion. It's not really meaningful to use tag [objectfifo], please change it to vectorization / matmul-elemenwise. Same for your other opening PRs.

**Comment by newling** [comment]
@Abhishek-Varma  What does this PR fix? I'm wondering if this should be done in https://github.com/nod-ai/iree-amd-aie/blob/main/compiler/plugins/target/AMD-AIE/aievec/VectorToVectorConversions.cpp where there are patterns for flattening vectors

**Pull request merged**